main.go

```diff
@@ -130,12 +130,36 @@ type SystemInfo struct {
 	CPUPhysicalCount int              `json:"cpu_physical_count"`
 }
 
+// SampleSummaryInt helps summarisation of uint64 based fields
+type SampleSummaryInt struct {
+	Maximum uint64  `json:"maximum"`
+	Mean    float64 `json:"mean"`
+	Minimum uint64  `json:"minimum"`
+}
+
+// SampleSummaryFloat helps summarisation of float64 based fields
+type SampleSummaryFloat struct {
+	Maximum float64 `json:"maximum"`
+	Mean    float64 `json:"mean"`
+	Minimum float64 `json:"minimum"`
+}
+
+// SampleSummary helps summarisation the more interesting fields.
+type SampleSummary struct {
+	CPUUser         SampleSummaryFloat `json:"cpu_user"`
+	CPUSystem       SampleSummaryFloat `json:"cpu_system"`
+	CPUIowait       SampleSummaryFloat `json:"cpu_iowait"`
+	RSS             SampleSummaryInt   `json:"rss"`
+	AvailableMemory SampleSummaryInt   `json:"available_memory"`
```

Comment:
I wonder if it's worth baking `Memory.UsedPercent` here as well, since a gauging percent might be a faster indicator for end-user in BQ than the overall counters for the available memory?